### PR TITLE
Modernize Repo (modules + golangci-lint)

### PR DESCRIPTION
**Pull request by keitwb**
- Got rid of vendor dir
- Got rid of Gopkg.* (dep stuff)
- Added go.mod
- Added golangci-lint test in Travis
- Ran goimports on all modules
- Replaced coverage assertion with standard go tools + grep
- Fixed up all of the golangci-lint failures
- Shortens CI time by about 5x.

### CODE REVIEW

**Comment by jgheewala** [comment]
@keitwb lets hold on to the commit until we have figured things out on our backend.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+run:
+  deadline: 1m
```
Comment (jgheewala) [inline .golangci.yml]: lets use timeout as deadline is marked for deprecation. I have made the changes in other repo

File: log/filter.go [hunk 2]
```diff
@@ -84,10 +85,7 @@ func (f *RegexFilter) WouldLog(keyvals ...interface{}) bool {
 		shouldPass = true
 	}
 	f.mu.RUnlock()
-	if shouldPass {
```
Comment (jgheewala) [inline log/filter.go]: heh

File: .golangci.yml (hunk 1 above)
Comment (jgheewala) [reply .golangci.yml]: https://github.com/golangci/golangci-lint/commit/98f60ebaa365593e40178b134a676adc7cb2aeaa

Comment (keitwb) [reply .golangci.yml]: Done

**Pull request merged**